src/legacy/core_plugins/input_control_vis/public/control/filter_manager/filter_manager.ts

```diff
@@ -19,35 +19,55 @@
 
 import _ from 'lodash';
 
-export class FilterManager {
+import { Filter } from 'src/plugins/data/common/es_query/filters';
+import { IndexPattern } from '../../legacy_imports';
+import { FilterManager as QueryFilterManager } from '../../../../../../plugins/data/public';
 
-  constructor(controlId, fieldName, indexPattern, queryFilter) {
+export abstract class FilterManager {
+  controlId: string;
+  fieldName: string;
+  indexPattern: IndexPattern;
+  queryFilter: QueryFilterManager;
+
+  constructor(
+    controlId: string,
+    fieldName: string,
+    indexPattern: IndexPattern,
+    queryFilter: QueryFilterManager
+  ) {
     this.controlId = controlId;
     this.fieldName = fieldName;
     this.indexPattern = indexPattern;
     this.queryFilter = queryFilter;
   }
 
-  getIndexPattern() {
+  /**
+   * Convert phrases into filter
+   *
+   * @param  {any[]} phrases
+   * @returns PhraseFilter
+   *   single phrase: match query
+   *   multiple phrases: bool query with should containing list of match_phrase queries
+   */
+  abstract createFilter(phrases: any): Filter;
+
+  abstract getValueFromFilterBar(): any;
+
+  getIndexPattern(): IndexPattern {
     return this.indexPattern;
   }
 
-  getField() {
+  getField(): any {
     return this.indexPattern.fields.getByName(this.fieldName);
   }
 
-  createFilter() {
-    throw new Error('Must implement createFilter.');
-  }
-
-  findFilters() {
-    const kbnFilters = _.flatten([this.queryFilter.getAppFilters(), this.queryFilter.getGlobalFilters()]);
-    return kbnFilters.filter((kbnFilter) => {
+  findFilters(): Filter[] {
+    const kbnFilters = _.flatten([
```

Comment:
I think you could avoid a lot of the complexity by passing in the `data` plugin, and then using `data.query.filterManager.getFilters()`

Reply:
Good point, however this PR I am trying to avoid functional changes and just shim the plugin.